### PR TITLE
feat(DataMapper): Add ChoiceSelectionService for xs:choice selection …

### LINKED ISSUE #2816: xs:choice: Apply Saved Selections
### Please describe the task that needs to be done

**Goal**: Load .kaoto config and apply selections to document model

### Changes

**File**: `packages/ui/src/services/choice-selection.service.ts`
Taking `field-type-override.service.ts` as a precedent, create a new `ChoiceSelectionService` as the orchestrator API collection

 - Key methods:
   - ~~`applyChoiceSelections(document, selections) `— apply saved selections to in-memory model~~ `DocumentUtilService.processOverrides()` should take care of that due to mutual interference among field type override, field substitution and choice
   - `setChoiceSelection(...)` — set selection + cascade-invalidate descendants (metadata + document)
   - `clearChoiceSelection(...)` — clear selection + cascade-invalidate
   - ~~`buildChoicePath(field)` — construct {choice:N} path from a field in the tree~~ We have introduced `SchemaPathService.build()` which could replace this
   - ~~`resolveChoicePath(document, choicePath)` — find choice field by {choice:N} path~~ We have introduced `SchemaPathService.navigateToChoiceField()` which could replace this
 - Path resolver walks the field tree matching `{choice:N} `segments to `isChoice` fields by positional index among sibling choices
 - Cascading invalidation: When setting a selection on path `/path/to/choice1`, remove all saved selections whose `choicePath` starts with `/path/to/choice1` and reset `selectedMemberIndex` on those descendant fields
 - When applying selections on load, sort by path depth (ascending) and skip unresolvable paths

**Testing**:
- Selections applied correctly
- Invalid paths handled gracefully
- Partial selections work

### PR DESCRIPTION
**Pull request by mmelko**
Fixes: #2816

## Summary

Implements the backend service for applying saved xs:choice selections during document initialization. This PR adds the [`ChoiceSelectionService`](packages/ui/src/services/choice-selection.service.ts:1) orchestrator and integrates it into the document creation pipeline, enabling users' choice selections to be persisted and automatically restored when documents are loaded.

### New Files
- **[`packages/ui/src/services/choice-selection.service.ts`](packages/ui/src/services/choice-selection.service.ts:1)** (264 lines)
  - Orchestrator service for xs:choice selection operations
  - Methods: [`applyChoiceSelections()`](packages/ui/src/services/choice-selection.service.ts:156), [`setChoiceSelection()`](packages/ui/src/services/choice-selection.service.ts:67), [`clearChoiceSelection()`](packages/ui/src/services/choice-selection.service.ts:109), [`buildChoicePath()`](packages/ui/src/services/choice-selection.service.ts:22), [`resolveChoicePath()`](packages/ui/src/services/choice-selection.service.ts:42)
  - Depth-ordered application (parent choices before children)
  - Graceful degradation for invalid paths/indices
  - Cascade invalidation of descendant selections

- **[`packages/ui/src/services/choice-selection.service.test.ts`](packages/ui/src/services/choice-selection.service.test.ts:1)** (501 lines)
  - 33 comprehensive test cases covering all service methods
  - Tests for path building, selection application, validation, and edge cases
  - All tests passing ✓

### Modified Files
- **[`packages/ui/src/services/field-type-override.service.ts`](packages/ui/src/services/field-type-override.service.ts:1)**
  - Removed deprecated `applyChoiceSelection()` and `revertChoiceSelection()` methods (75 lines)
  - Removed unused `IChoiceSelection` import
  - Updated documentation to reference [`ChoiceSelectionService`](packages/ui/src/services/choice-selection.service.ts:1)

- **[`packages/ui/src/services/field-type-override.service.test.ts`](packages/ui/src/services/field-type-override.service.test.ts:1)**
  - Removed 107 lines of test cases for deprecated methods
  - All remaining 42 tests passing ✓

## Summary by CodeRabbit

* **New Features / Refactor**
  * Introduced a dedicated choice-selection service and moved choice-selection responsibilities for clearer ownership.

* **Bug Fixes**
  * Avoid redundant choice updates (idempotent apply).
  * Ensure setting/clearing selections cascades invalidation to nested choices while preserving unrelated sibling subtrees.
  * Validate inputs and surface errors for non-choice fields and invalid member indices.

* **Tests**
  * Added tests for set/clear/idempotency and descendant-invalidation; removed overlapping legacy tests.

### CODE REVIEW

File: packages/ui/src/services/choice-selection.service.ts [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+   * @example
+   * ```typescript
+   * const selections = [
+   *   { schemaPath: '/ns0:Root/{choice:0}', selectedMemberIndex: 1 },
+   *   { schemaPath: '/ns0:Root/{choice:0}/ns0:Option1/{choice:0}', selectedMemberIndex: 0 }
+   * ];
+   * ChoiceSelectionService.applyChoiceSelections(document, selections, namespaceMap);
+   * ```
+   */
+  static applyChoiceSelections(
```
Comment (igarashitm) [inline packages/ui/src/services/choice-selection.service.ts]: I think this is a duplicate of `DocumentUtilService.processOverrides()`, issue description is out of date due to the fix for handling mutual interference among field type override, field substitution and choice. I'll update the issue.
Then here we need to
1. Remove this method
2. Make sure DocumentUtilService.processOverrides() satisfies the requirements `applyChoiceSelections()` was supposed to provide
3. Migrate tests to document-util.service.test.ts with invoking `processOverrides()` if there're missing cases

File: packages/ui/src/services/choice-selection.service.ts [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+    const selection: IChoiceSelection = { schemaPath, selectedMemberIndex };
+
+    // Apply to field and definition
+    const changed = DocumentUtilService.processChoiceSelection(document, selection, namespaceMap);
+
+    // Cascade invalidate descendants
+    if (changed) {
+      this.resetDescendantFields(document, schemaPath, namespaceMap);
+      DocumentUtilService.invalidateDescendants(document, schemaPath);
+    }
```
[the remark this answers is not in the document]
Comment (igarashitm) [reply packages/ui/src/services/choice-selection.service.ts:80]: This sounds like a valid concern, let's add an early return with `false` if selectedMemberIndex is already set and is setting a same index, but probably in `DocumentUtilService.applyChoiceSelectionToField()`

File: packages/ui/src/services/choice-selection.service.ts [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+   *
+   * @example
+   * ```typescript
+   * const path = ChoiceSelectionService.buildChoicePath(choiceField, namespaceMap);
+   * // Returns: '/ns0:Root/{choice:0}'
+   * ```
+   *
+   * @see resolveChoicePath
+   */
+  static buildChoicePath(choiceField: IField, namespaceMap: Record<string, string>): string {
```
Comment (igarashitm) [inline packages/ui/src/services/choice-selection.service.ts]: Also an out of date issue description, as it looks we have introduced `SchemaPathService.build()` which covers the use case. Let's remove it and migrate the test cases that covers new cases

File: packages/ui/src/services/choice-selection.service.ts [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+   *   namespaceMap
+   * );
+   * if (field) {
+   *   console.log('Found choice with', field.fields.length, 'members');
+   * }
+   * ```
+   *
+   * @see buildChoicePath
+   */
+  static resolveChoicePath(
```
Comment (igarashitm) [inline packages/ui/src/services/choice-selection.service.ts]: Also an out of date issue description, as it looks we have introduced `SchemaPathService.navigateToChoiceField()` which covers the use case. Let's remove it and migrate the test cases that covers new cases

**Review by igarashitm — APPROVED** [review]
